ibis/backends/pyspark/__init__.py

```diff
@@ -514,7 +515,14 @@ def get_schema(
         table_loc = self._to_sqlglot_table((catalog, database))
         catalog, db = self._to_catalog_db_tuple(table_loc)
         with self._active_catalog_database(catalog, db):
-            df = self._session.table(table_name)
+            try:
+                df = self._session.table(table_name)
+            except AnalysisException as e:
+                # we can alternative check for e.getErrorClass == 'TABLE_OR_VIEW_NOT_FOUND'
+                if e.getSqlState() == "42P01":
+                    # spark sqlerror state for undefined_table see https://spark.apache.org/docs/3.4.1/sql-error-conditions-sqlstates.html
```

Comment:
I think there are less chances that the code name, but we can check for `e.getErrorClass == 'TABLE_OR_VIEW_NOT_FOUND'` or both, suggestions welcome.

Reply:
Checking the error class seems much more maintainable.